scripts/train/train.py

```diff
@@ -134,224 +243,100 @@ def main(cfg: DictConfig) -> Trainer:
-    validate_config(cfg)
-
-    # Resolve all interpolation variables as early as possible
-    om.resolve(cfg)
-
-    # Create copy of config for logging
-    logged_cfg: DictConfig = copy.deepcopy(cfg)
+    validate_config(scfg)
 
     cuda_alloc_conf = []
     # Get max split size mb
-    max_split_size_mb: Optional[int] = cfg.pop('max_split_size_mb', None)
+    max_split_size_mb: Optional[int] = scfg.max_split_size_mb
     if max_split_size_mb is not None:
         cuda_alloc_conf.append(f'max_split_size_mb:{max_split_size_mb}')
 
     # Expandable segments
-    if cfg.pop('expandable_segments', False):
+    if scfg.expandable_segments:
         cuda_alloc_conf.append('expandable_segments:True')
 
     if len(cuda_alloc_conf) > 0:
         os.environ['PYTORCH_CUDA_ALLOC_CONF'] = ','.join(cuda_alloc_conf)
 
     # Set CUDA lazy loading
     # This can save a bit of memory if not all modules are needed
-    cuda_load_lazy: bool = cfg.pop('cuda_load_lazy', False)
+    cuda_load_lazy: bool = scfg.cuda_load_lazy
     if cuda_load_lazy:
         os.environ['CUDA_MODULE_LOADING'] = 'LAZY'
 
     # Set seed first
-    seed: int = pop_config(cfg, 'seed', must_exist=True)
+    seed: int = scfg.seed
     reproducibility.seed_all(seed)
 
     # Initialize pytorch distributed training process groups
-    dist_timeout: Union[int, float] = pop_config(cfg,
-                                                 'dist_timeout',
-                                                 must_exist=False,
-                                                 default_value=600.0)
+    dist_timeout: Union[int, float] = scfg.dist_timeout
     dist.initialize_dist(get_device(None), timeout=dist_timeout)
 
-    # Get global and device batch size information from distributed/single node setting
-    cfg = update_batch_size_info(cfg)
-    logged_cfg.update(cfg, merge=True)
-
     # Mandatory model training configs
-    model_config: DictConfig = pop_config(cfg, 'model', must_exist=True)
-    tokenizer_config: Dict[str, Any] = pop_config(cfg,
-                                                  'tokenizer',
-                                                  must_exist=True,
-                                                  convert=True)
-    optimizer_config: Dict[str, Any] = pop_config(cfg,
-                                                  'optimizer',
-                                                  must_exist=True,
-                                                  convert=True)
-    scheduler_config: Dict[str, Any] = pop_config(cfg,
-                                                  'scheduler',
-                                                  must_exist=True,
-                                                  convert=True)
-    train_loader_config: DictConfig = pop_config(cfg,
-                                                 'train_loader',
-                                                 must_exist=True)
+    model_config: DictConfig = DictConfig(scfg.model)
```

Comment:
Can you add separate helper functions for each of these config "sections"? i.e. Mandatory model training configs, Optional fsdp data, fine-tuning, and eval configs, etc.

Reply:
I reorganized the `train` and `eval` main functions for clarity, let me know if this suffices - I'm skeptical of helper functions here because so much of this setup is building variables for shared state, so it seems like we might be at risk of replacing

```
  x_config = train_config.x or environment_default(yada)
  y_config = train_config.y or environment_default(thing2)
  ...
```

with 

```
   x_config, y_config = _internal_get_config_helper(z_config, w_config, a_config, b_config, ...)
```

which may not be that much clearer

Reply:
actually scratch that haha I think there's plenty of room for abstracting out helpers